### PR TITLE
Add ability to hide quit messages of K/G/Z:lined users

### PR DESCRIPTION
**Pull request by Zoddo**
This add `set::options::secret-ban-reasons` in the configuration file

Suggested by Aubrey ([#3993](https://bugs.unrealircd.org/view.php?id=3993))

### CODE REVIEW

File: doc/conf/examples/example.conf [hunk 1]
```diff
@@ -390,6 +390,7 @@ set {
 	options {
 		hide-ulines; /* hide U-lines in /MAP and /LINKS */
 		show-connect-info; /* show "looking up your hostname" messages on connect */
+		/* secret-ban-reasons; don't publicely display K/G/Z:lines reasons (in the QUIT message) */
```
Comment (GLolol) [inline doc/conf/examples/example.conf]: Publicly, not public**e**ly.

Comment (Zoddo) [inline doc/conf/examples/example.conf]: Oops, corrected :)

**Comment by dboyz** [comment]
I might be wrong but I think you may want to initialize `secret_ban_reasons` somewhere.

**Comment by Zoddo** [comment]
It seem there isn't other code for the other config options (I [checked](https://github.com/unrealircd/unrealircd/search?utf8=%E2%9C%93&q=allow_insane_bans) `allow_insane_bans`)

**Comment by syzop** [comment]
See https://bugs.unrealircd.org/view.php?id=3993 for discussion

**Pull request closed without merging**